### PR TITLE
fix: add docs version navbar

### PR DESCRIPTION
**Pull request by yoonhyejin**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

This PR Adds versioning dropdown to the docs-site navbar.

The versioning dropdown will link to the older Vercel deployments that correspond to specific release versions, based on the recommendation from Docusaurus docs (source: https://docusaurus.io/docs/versioning#keep-the-number-of-versions-small). 

This method was chosen to/because 
- avoid complicating the contribution process and 
- avoid the need for a complete refactor on the asset path. 
- we might not actively edit older versions of the docs (although it's still possible to edit them if necessary) 

As this is the first time we are utilizing the versioning dropdown, the "older versions" will not have dropdowns initially.

Runbook for Future Versioning: (TBD)
For future versioning, we would: 
- Edit the "label" of the dropdown to the latest version. (ex. v0.10.5) 
- Add the last version (ex. v0.10.4)  to the dropdown list and link it to the corresponding Vercel deployment.

Resolves DR-465

### CODE REVIEW

File: docs-website/docusaurus.config.js [hunk 1]
```diff
@@ -69,6 +71,26 @@ module.exports = {
           label: "Roadmap",
           position: "right",
         },
+        {
+          type: "dropdown",
+          label: "v0.10.4",
```
Comment (hsheth2) [inline docs-website/docusaurus.config.js:76]: this should always be "latest"

File: docs-website/docusaurus.config.js [hunk 2]
```diff
@@ -69,6 +71,26 @@ module.exports = {
...
+          position: "right",
+          items: [
+            {
+              label: "v0.10.3",
+              href: "https://docs-website-lgy6es9n7-acryldata.vercel.app/docs/",
```
Comment (hsheth2) [inline docs-website/docusaurus.config.js:81]: what's the plan for keeping these up to date? does this need to be part of our release process?

Comment (yoonhyejin) [reply docs-website/docusaurus.config.js:81]: I was thinking of linking these to specific vercel deployments that correspond to the last commit of each release - would that work?

Comment (hsheth2) [reply docs-website/docusaurus.config.js:81]: Yup, let’s just make sure to update our docs to include this step in the release process

File: docs-website/docusaurus.config.js (hunk 1 above)
Comment (hsheth2) [reply docs-website/docusaurus.config.js:76]: Or ideally - when building from vercel, it has the version number, and when building as part of our docs generation, it says “latest”

**Comment by yoonhyejin** [comment]
changing this to a draft since I'm going to test the native versioning with the static-assets repo, based on the feedback below: (cc. @hsheth2 @maggiehays ) 

1. We don’t have a good plan to handle RCs for docs ("Next version" ...)  
2. The site is being redirected to the main page everything the user changes version

**Pull request closed without merging**